src/java/org/apache/cassandra/io/compress/CompressionMetadata.java

```diff
@@ -204,6 +206,7 @@ private CompressionMetadata(File filePath, CompressionParams parameters, Memory.
         this.compressedFileLength = compressedLength;
         assert offsets != null;
         this.chunkOffsets = offsets;
+        this.owningChunkOffsets = false;
```

Comment:
This is severely undercounting the memory in use for classic C* nodes, where most of the `CompressionMetadata` objects will come from the result of compaction. In these cases `open` is given something that was prepared by the writer, but it holds on to it for the lifetime of the sstable reader.

I think the easiest safe method to solve this is to double-count the usage for a while, i.e. add when we do `sharedCopy`. The duration that this overcounting is in effect should be very short.

To do better than that we may need to expose a boolean return on `SafeMemory.free()` that only returns true if the global reference is released.